schemas/webpackOptionsSchema.json

```diff
@@ -0,0 +1,478 @@
+      "oneOf": [
+        {
+          "description": "Multiple entry bundles are created. The key is the chunk name. The value can be a string or an array.",
+          "type": "object"
+        },
+        {
+          "description": "The string is resolved to a module which is loaded upon startup.",
+          "type": "string"
+        },
+        {
+          "allOf": [
+          	{
+	            "ref": "#/definitions/common.nonEmptyArrayOfUniqueStringValues"
+	          }
+          ],
+          "description": "All modules are loaded upon startup. The last one is exported."
+        }
+      ]
+    },
+    "externals": {
+      "description": "Specify dependencies that shouldn't be resolved by webpack, but should become dependencies of the resulting bundle. The kind of the dependency depends on `output.libraryTarget`.",
+      "oneOf": [
+        {
+          "description": "An exact matched dependency becomes external. The same string is used as external dependency.",
+          "type": "string"
+        },
+        {
+          "description": "If an dependency matches exactly a property of the object, the property value is used as dependency.",
+          "type": "object"
+        },
+        {
+          "description": "`function(context, request, callback(err, result))` The function is called on each dependency.",
+          "format": "function"
+        },
+        {
+          "description": "Every matched dependency becomes external.",
+          "format": "regex"
+        },
+        {
+          "items": {
+            "$ref": "#/definitions/externals"
+          },
+          "type": "array"
+        }
+      ]
+    },
+    "module": {
+      "description": "Options affecting the normal modules (`NormalModuleFactory`).",
+      "properties": {
+        "loaders": {
+          "allOf": [
+          	{
+	            "$ref": "#/definitions/common.loaders"
+	          }
+          ],
+          "description": "An array of automatically applied loaders."
+        },
+        "noParse": {
+          "description": "Don't parse files matching. It's matched against the full resolved request.",
+          "oneOf": [
+            {
+              "items": {
+                "format": "regex"
+              },
+              "minItems": 1,
+              "type": "array",
+              "uniqueItems": true
+            },
+            {
+              "format": "regex"
+            }
+          ]
+        },
+        "postLoaders": {
+          "allOf": [
+          	{
+	            "$ref": "#/definitions/common.loaders"
+	          }
+          ],
+          "description": "An array of automatically applied loaders."
+        },
+        "preLoaders": {
+          "allOf": [
+          	{
+	            "$ref": "#/definitions/common.loaders"
+	          }
+          ],
+          "description": "An array of automatically applied loaders."
+        }
+      },
+      "type": "object"
+    },
+    "output": {
```

Comment:
Missing auxiliaryComments, heres the PR for it. 

I hadn't gotten to documenting it yet. :eyes: 

https://github.com/webpack/webpack/pull/2113

It is essentially: 
`auxiliaryComments: string|object`

If `object` properties are: 
```
{
  root?: string
  commonjs?: string
  commonjs2?: string
  amd?: string
}
```